### PR TITLE
Add Acra

### PR DESCRIPTION
**Pull request by sytolk**
https://github.com/pires/android-obd-reader/issues/135
Not tested yet.. and I`m not sure that openshift support php mail function :)

### CODE REVIEW

File: src/main/java/com/github/pires/obd/reader/ObdReaderApplication.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.HashMap;
+
+/**
+ * Created with IntelliJ IDEA.
+ * User: stanimir
+ * Date: 9/1/15
+ * Time: 5:31 PM
+ * developer STANIMIR MARINOV
+ */
```
Comment (pires) [inline src/main/java/com/github/pires/obd/reader/ObdReaderApplication.java]: Remove these comments.

File: src/main/java/com/github/pires/obd/reader/acra/ACRAPostSender.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import org.apache.http.protocol.HTTP;
+
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.*;
+
+
+public class ACRAPostSender implements ReportSender {
+    private final static String BASE_URL = "http://acra-rest1.rhcloud.com/acra.php?email=smarinov@abv.bg";
```
Comment (pires) [inline src/main/java/com/github/pires/obd/reader/acra/ACRAPostSender.java]: This should be configurable to provide this feature to other users.

File: src/main/java/com/github/pires/obd/reader/acra/ACRAPostSender.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    private final static String SHARED_SECRET = "my_shared_secret";
```
Comment (pires) [inline src/main/java/com/github/pires/obd/reader/acra/ACRAPostSender.java]: This should be configurable to provide this feature to other users.

**Comment by pires** [comment]
You're missing `build.gradle` as well.

File: src/main/java/com/github/pires/obd/reader/acra/ACRAPostSender.java (hunk 3 above)
Comment (sytolk) [inline src/main/java/com/github/pires/obd/reader/acra/ACRAPostSender.java]: SHARED_SECRET must be the same like in acra.php
My idea was to use only one acra.php script hosted on openshift and only user email can be configurable. Application can have 1 hardcoded default email.. or better email=null if(email != null) ACRA will send log to this email.

**Comment by sytolk** [comment]
The main problem.. its appears that openshift will not send emails. https://forums.openshift.com/php-53-of-openshift-they-support-the-mail-functions

Its need to have one hardcoded google acount (user/pass) and use googles SMTP to send emails.. I not like this solution

**Comment by sytolk** [comment]
I have tested report sending and it works. I have use mu mailgun key to send emails. I will publish PHP code too.

**Comment by pires** [comment]
While it's important to you, it's not ready for generic usage. I'm sorry but this won't be merged as is.

**Comment by sytolk** [comment]
Ok :)

**Comment by sytolk** [comment]
@pires I have write TODO for the SHARED_SECRET. You can merge it now (debugging works) and to open new issue with this.. actually SHARED_SECRET is for the security but not for the obd-reader application this is for the security for the php mailer. But this can be done later. I`m to lazy to finish it now ;)

**Comment by pires** [comment]
@sytolk I understand laziness, I'm lazy too. But we need variables to be configurable. All of them have been detailed before.
Also, the commit log is a mess. Please meld your commits into a single one after rebasing current changes in `master`.

**Comment by sytolk** [comment]
@pires you are right.. you are lazy too :) maybe I will do this but my working priorities are different now and I really want to see this PR https://github.com/pires/android-obd-reader/pull/144 finished. good luck @Pho3niX90 :+1:

**Pull request closed without merging**